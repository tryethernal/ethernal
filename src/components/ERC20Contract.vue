<template>
    <v-container fluid>
        <v-row class="mb-1">
            <v-col cols="12" lg="5">
                <v-card outlined style="height: 100%">
                    <v-skeleton-loader v-if="loadingContract" type="list-item"></v-skeleton-loader>
                    <v-card-title v-else>{{ contract.tokenName }}</v-card-title>
                    <v-card-subtitle>
                        <v-skeleton-loader v-if="loadingContract" type="chip"></v-skeleton-loader>
                        <v-chip v-else v-for="(pattern, idx) in contract.patterns" :key="idx" x-small class="success mr-2">
                            {{ formatContractPattern(pattern) }}
                        </v-chip>
                    </v-card-subtitle>
                    <v-card-text>
                        <v-row>
                            <v-col cols="6">
                                <small>Total Supply</small><br>
                                <v-skeleton-loader v-if="loadingContract" type="list-item"></v-skeleton-loader>
                                <span class="text-h6 ml-2" v-else-if="contract.tokenTotalSupply">{{ formatNumber(contract.tokenTotalSupply, { decimals: contract.tokenDecimals }) }} {{ contract.tokenSymbol }}</span>
                                <span class="text-h6 ml-2" v-else>N/A</span>
                            </v-col>

                            <v-col cols="6">
                                <small>Decimals</small><br>
                                <v-skeleton-loader v-if="loadingContract" type="list-item"></v-skeleton-loader>
                                <span v-else class="text-h6 ml-2">{{ contract.tokenDecimals || 'N/A' }}</span>
                            </v-col>
                        </v-row>

                        <v-row>
                            <v-col cols="6">
                                <small>Contract</small><br>
                                <v-skeleton-loader v-if="loadingContract" type="list-item"></v-skeleton-loader>
                                <Hash-Link class="ml-2" :type="'contract'" :hash="contract.address" :withName="true" />
                            </v-col>

                            <v-col cols="6">
                                <small>Contract Creation</small><br>
                                <v-skeleton-loader v-if="loadingContract" type="list-item"></v-skeleton-loader>
                                <span v-else-if="contract.creationTransaction && contract.creationTransaction.hash" class="ml-2">
                                    <Hash-Link :type="'transaction'" :hash="contract.creationTransaction.hash" />
                                </span>
                                <span v-else class="text-h6 ml-2">N/A</span>
                            </v-col>

                        </v-row>
                    </v-card-text>
                </v-card>
            </v-col>

            <v-col cols="12" lg="7">
                <v-row>
                    <v-col cols="12" sm="6" lg="6">
                        <Stat-Number :loading="loadingStats" :title="'Holders'" :value="contractStats.tokenHolderCount" />
                    </v-col>

                    <v-col cols="12" sm="6" lg="6">
                        <Stat-Number :loading="loadingStats" :title="'Transfers'" :value="contractStats.tokenTransferCount" />
                    </v-col>

                    <v-col cols="12" sm="6" lg="6">
                        <Stat-Number :loading="loadingStats" :title="'Circulating Supply'" :value="contractStats.tokenCirculatingSupply" :decimals="contract.tokenDecimals" :infoTooltip="'Number of minted tokens'" />
                    </v-col>

                    <v-col cols="12" sm="6" lg="6">
                        <v-card outlined style="height: 100%">
                            <v-card-subtitle v-if="metamaskData.account && metamaskData.isReady">
                                <div style="position: absolute;">Your Balance</div>
                                <div class="text-right" v-if="metamaskData.account">
                                    <Hash-Link :type="'address'" :hash="metamaskData.account" />
                                </div>
                            </v-card-subtitle>
                            <v-card-subtitle v-else>Your Balance</v-card-subtitle>
                            <v-card-text class="text-h3" align="center" v-if="metamaskData.account && metamaskData.isReady">
                                <v-skeleton-loader v-if="loadingBalance" type="list-item"></v-skeleton-loader>
                                <template v-else-if="connectedAccountBalance">{{ formatNumber(connectedAccountBalance, { short: true }) }} {{ contract.tokenSymbol }}</template>
                                <template v-else>N/A</template>
                            </v-card-text>
                            <v-card-text v-else>
                                <Metamask class="mt-1" @rpcConnectionStatusChanged="onRpcConnectionStatusChanged"></Metamask>
                            </v-card-text>
                        </v-card>
                    </v-col>
                </v-row>
            </v-col>
        </v-row>

        <v-tabs v-model="tab">
            <v-tab id="transactionsTab" href="#transactions">Transactions</v-tab>
            <v-tab id="interactionsTab" href="#interactions">Read / Write</v-tab>
            <v-tab id="holdersTab" href="#holders">Holders</v-tab>
            <v-tab id="transfersTab" href="#transfers">Transfers</v-tab>
            <v-tab style="display: none;" id="analyticsTab" href="#analytics">Analytics</v-tab>
        </v-tabs>

        <v-tabs-items :value="tab">
            <v-tab-item value="transactions">
                <Address-Transactions-List :address="address" />
            </v-tab-item>

            <v-tab-item value="interactions">
                <Contract-Interaction :address="address" />
            </v-tab-item>

            <v-tab-item value="holders">
                <ERC-20-Token-Holders :address="address" :tokenDecimals="contract.tokenDecimals" :tokenSymbol="contract.tokenSymbol" />
            </v-tab-item>

            <v-tab-item value="analytics">
                <ERC-20-Contract-Analytics :address="address" :tokenDecimals="contract.tokenDecimals" :tokenSymbol="contract.tokenSymbol" />
            </v-tab-item>

            <v-tab-item value="transfers">
                <ERC-20-Token-Transfers :address="address" :tokenDecimals="contract.tokenDecimals" :tokenSymbol="contract.tokenSymbol" />
            </v-tab-item>
        </v-tabs-items>
    </v-container>
</template>

<script>
const moment = require('moment');
import { mapGetters } from 'vuex';

const { formatNumber, formatContractPattern } = require('../lib/utils');
const ERC20_ABI = require('../abis/erc20.json');

import AddressTransactionsList from './AddressTransactionsList';
import ContractInteraction from './ContractInteraction';
import ERC20TokenHolders from './ERC20TokenHolders';
import ERC20ContractAnalytics from './ERC20ContractAnalytics';
import ERC20TokenTransfers from './ERC20TokenTransfers';
import StatNumber from './StatNumber';
import HashLink from './HashLink';
import Metamask from './Metamask';

export default {
    name: 'ERC20Contract',
    props: ['address'],
    components: {
        AddressTransactionsList,
        StatNumber,
        HashLink,
        Metamask,
        ContractInteraction,
        ERC20TokenHolders,
        ERC20ContractAnalytics,
        ERC20TokenTransfers
    },
    data: () => ({
        loadingContract: true,
        loadingStats: true,
        loadingBalance: false,
        contract: {},
        contractStats: {},
        metamaskData: {},
        connectedAccountBalance: null,
    }),
    mounted() {
    },
    methods: {
        moment: moment,
        formatNumber: formatNumber,
        formatContractPattern: formatContractPattern,
        onRpcConnectionStatusChanged(data) {
            this.metamaskData = data;
            if (data.account && data.isReady) {
                this.loadingBalance = true;
                this.server.callContractReadMethod(
                    { address: this.address, abi: ERC20_ABI },
                    'balanceOf(address)',
                    { from: null },
                    { 0: data.account },
                    this.currentWorkspace.rpcServer,
                    window.ethereum
                )
                .then(([balance]) => this.connectedAccountBalance = balance)
                .finally(() => this.loadingBalance = false);
            }
            else
                this.connectedAccountBalance = null;
        },
    },
    watch: {
        address: {
            immediate: true,
            handler(address) {
                this.server.getContract(address)
                    .then(({ data }) => this.contract = data)
                    .finally(() => this.loadingContract = false);

                this.server.getContractStats(address)
                    .then(({ data }) => this.contractStats = data)
                    .finally(() => this.loadingStats = false);
            }
        }
    },
    computed: {
        ...mapGetters([
            'currentWorkspace'
        ]),
        tab: {
            set(tab) {
                this.$router.replace({ query: { ...this.$route.query, tab } }).catch(()=>{});
            },
            get() {
                return this.$route.query.tab;
            }
        }
    }
}
</script>
