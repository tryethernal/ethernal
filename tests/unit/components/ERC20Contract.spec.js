import MockHelper from '../MockHelper';
import flushPromises from 'flush-promises';
import ERC20Contract from '@/components/ERC20Contract.vue';

let helper;
const stubs = [
    'Hash-Link',
    'Stat-Number',
    'Contract-Interaction',
    'ERC-2O-Token-Holders',
    'ERC-2O-Contract-Analytics',
    'ERC-2O-Token-Transfers',
    'Metamask'
];

describe('ERC20Contract.vue', () => {

    beforeEach(() => {
        jest.clearAllMocks();
        helper = new MockHelper();
    });

    it('Should display contract info', async () => {
        jest.spyOn(helper.mocks.server, 'getContract')
            .mockResolvedValueOnce({ data: {
                tokenName: 'Amalfi',
                patterns: ['erc20'],
                tokenTotalSupply: '1000000000',
                tokenDecimals: 2,
                address: '0x123',
                creationTransaction: { hash: '0xabc' }
            }});

        jest.spyOn(helper.mocks.server, 'getContractStats')
            .mockResolvedValueOnce({ data: {
                tokenHolderCount: 1,
                tokenTransferCount: 2,
                tokenCirculatingSupply: '1000000000',
            }});

        const wrapper = helper.mountFn(ERC20Contract, {
            propsData: {
                address: '0x123'
            },
            stubs: stubs
        });

        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should display placeholders', async () => {
        jest.spyOn(helper.mocks.server, 'getContract')
            .mockResolvedValueOnce({ data: {
                name: 'Amalfi',
                tokenName: null,
                patterns: [],
                tokenTotalSupply: null,
                tokenDecimals: null,
                address: '0x123',
                creationTransaction: { hash: '0xabc' }
            }});

        jest.spyOn(helper.mocks.server, 'getContractStats')
            .mockResolvedValueOnce({ data: {
                tokenHolderCount: 0,
                tokenTransferCount: 0,
                tokenCirculatingSupply: 0,
            }});

        const wrapper = helper.mountFn(ERC20Contract, {
            propsData: {
                address: '0x123'
            },
            stubs: stubs
        });

        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot();
    });
});
