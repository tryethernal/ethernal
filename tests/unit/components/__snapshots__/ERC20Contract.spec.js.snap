// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ERC20Contract.vue Should display contract info 1`] = `
<div class="container container--fluid">
  <div class="row mb-1">
    <div class="col-lg-5 col-12">
      <div class="v-card v-sheet v-sheet--outlined theme--light" style="height: 100%;">
        <div class="v-card__title">Amalfi</div>
        <div class="v-card__subtitle"><span class="success mr-2 v-chip v-chip--no-color theme--light v-size--x-small"><span class="v-chip__content">
                        ERC 20
                    </span></span></div>
        <div class="v-card__text">
          <div class="row">
            <div class="col col-6"><small>Total Supply</small><br> <span class="text-h6 ml-2">10,000,000 </span></div>
            <div class="col col-6"><small>Decimals</small><br> <span class="text-h6 ml-2">2</span></div>
          </div>
          <div class="row">
            <div class="col col-6"><small>Contract</small><br>
              <!---->
              <hash-link-stub withname="true" type="contract" class="ml-2" hash="0x123"></hash-link-stub>
            </div>
            <div class="col col-6"><small>Contract Creation</small><br> <span class="ml-2"><hash-link-stub hash="0xabc" type="transaction"></hash-link-stub></span></div>
          </div>
        </div>
      </div>
    </div>
    <div class="col-lg-7 col-12">
      <div class="row">
        <div class="col-sm-6 col-lg-6 col-12">
          <stat-number-stub title="Holders" value="1"></stat-number-stub>
        </div>
        <div class="col-sm-6 col-lg-6 col-12">
          <stat-number-stub title="Transfers" value="2"></stat-number-stub>
        </div>
        <div class="col-sm-6 col-lg-6 col-12">
          <stat-number-stub infotooltip="Number of minted tokens" title="Circulating Supply" decimals="2" value="1000000000"></stat-number-stub>
        </div>
        <div class="col-sm-6 col-lg-6 col-12">
          <div class="v-card v-sheet v-sheet--outlined theme--light" style="height: 100%;">
            <div class="v-card__subtitle">Your Balance</div>
            <div class="v-card__text">
              <metamask-stub class="mt-1"></metamask-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="v-tabs theme--light">
    <div role="tablist" class="v-item-group theme--light v-slide-group v-tabs-bar v-tabs-bar--is-mobile primary--text">
      <div class="v-slide-group__prev v-slide-group__prev--disabled">
        <transition-stub name="fade-transition"></transition-stub>
      </div>
      <div class="v-slide-group__wrapper">
        <div class="v-slide-group__content v-tabs-bar__content">
          <div class="v-tabs-slider-wrapper" style="transition: none;">
            <div class="v-tabs-slider"></div>
          </div><a tabindex="0" href="#transactions" aria-selected="true" role="tab" class="v-tab v-tab--active" id="transactionsTab">Transactions</a> <a tabindex="0" href="#interactions" aria-selected="false" role="tab" class="v-tab" id="interactionsTab">Read / Write</a> <a tabindex="0" href="#holders" aria-selected="false" role="tab" class="v-tab" id="holdersTab">Holders</a> <a tabindex="0" href="#transfers" aria-selected="false" role="tab" class="v-tab" id="transfersTab">Transfers</a> <a tabindex="0" href="#analytics" aria-selected="false" role="tab" class="v-tab" style="display: none;" id="analyticsTab">Analytics</a>
        </div>
      </div>
      <div class="v-slide-group__next v-slide-group__next--disabled">
        <transition-stub name="fade-transition"></transition-stub>
      </div>
    </div>
  </div>
  <div class="v-window v-item-group theme--light v-tabs-items">
    <div class="v-window__container">
      <transition-stub name="">
        <div class="v-window-item v-window-item--active" id="transactions">
          <div class="container container--fluid">
            <div class="v-data-table v-data-table--has-bottom theme--light">
              <div class="v-data-table__wrapper">
                <table>
                  <colgroup>
                    <col class="">
                    <col class="">
                    <col class="">
                    <col class="">
                    <col class="">
                    <col class="">
                    <col class="">
                    <col class="">
                  </colgroup>
                  <thead class="v-data-table-header">
                    <tr>
                      <th role="columnheader" scope="col" aria-label="Txn Hash: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Txn Hash</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="Method" class="text-start"><span>Method</span></th>
                      <th role="columnheader" scope="col" aria-label="Block: Sorted descending. Activate to sort ascending." aria-sort="descending" class="text-start sortable active desc"><span>Block</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="Mined On: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Mined On</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="From: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>From</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="To: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>To</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="Value: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Value</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="Fee" class="text-start"><span>Fee</span></th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr class="v-data-table__empty-wrapper">
                      <td colspan="8">
                        No transactions found
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <div class="v-data-footer">
                <div class="v-data-footer__select">Rows per page:<div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select">
                    <div class="v-input__control">
                      <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-47" class="v-input__slot">
                        <div class="v-select__slot">
                          <div class="v-select__selections">
                            <div class="v-select__selection v-select__selection--comma">10</div><input aria-label="Rows per page:" id="input-47" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                          </div>
                          <div class="v-input__append-inner">
                            <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                          </div><input type="hidden" value="10">
                        </div>
                        <div class="v-menu">
                          <!---->
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="v-data-footer__pagination">â€“</div>
                <div class="v-data-footer__icons-before"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Previous page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--light"></i></span></button></div>
                <div class="v-data-footer__icons-after"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Next page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light"></i></span></button></div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <transition-stub name="">
        <!---->
      </transition-stub>
      <transition-stub name="">
        <!---->
      </transition-stub>
      <transition-stub name="">
        <!---->
      </transition-stub>
      <transition-stub name="">
        <!---->
      </transition-stub>
    </div>
  </div>
</div>
`;

exports[`ERC20Contract.vue Should display placeholders 1`] = `
<div class="container container--fluid">
  <div class="row mb-1">
    <div class="col-lg-5 col-12">
      <div class="v-card v-sheet v-sheet--outlined theme--light" style="height: 100%;">
        <div class="v-card__title"></div>
        <div class="v-card__subtitle"></div>
        <div class="v-card__text">
          <div class="row">
            <div class="col col-6"><small>Total Supply</small><br> <span class="text-h6 ml-2">N/A</span></div>
            <div class="col col-6"><small>Decimals</small><br> <span class="text-h6 ml-2">N/A</span></div>
          </div>
          <div class="row">
            <div class="col col-6"><small>Contract</small><br>
              <!---->
              <hash-link-stub withname="true" type="contract" class="ml-2" hash="0x123"></hash-link-stub>
            </div>
            <div class="col col-6"><small>Contract Creation</small><br> <span class="ml-2"><hash-link-stub hash="0xabc" type="transaction"></hash-link-stub></span></div>
          </div>
        </div>
      </div>
    </div>
    <div class="col-lg-7 col-12">
      <div class="row">
        <div class="col-sm-6 col-lg-6 col-12">
          <stat-number-stub title="Holders" value="0"></stat-number-stub>
        </div>
        <div class="col-sm-6 col-lg-6 col-12">
          <stat-number-stub title="Transfers" value="0"></stat-number-stub>
        </div>
        <div class="col-sm-6 col-lg-6 col-12">
          <stat-number-stub infotooltip="Number of minted tokens" title="Circulating Supply" value="0"></stat-number-stub>
        </div>
        <div class="col-sm-6 col-lg-6 col-12">
          <div class="v-card v-sheet v-sheet--outlined theme--light" style="height: 100%;">
            <div class="v-card__subtitle">Your Balance</div>
            <div class="v-card__text">
              <metamask-stub class="mt-1"></metamask-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="v-tabs theme--light">
    <div role="tablist" class="v-item-group theme--light v-slide-group v-tabs-bar v-tabs-bar--is-mobile primary--text" data-booted="true">
      <div class="v-slide-group__prev v-slide-group__prev--disabled">
        <transition-stub name="fade-transition"></transition-stub>
      </div>
      <div class="v-slide-group__wrapper">
        <div class="v-slide-group__content v-tabs-bar__content">
          <div class="v-tabs-slider-wrapper" style="transition: none;">
            <div class="v-tabs-slider"></div>
          </div><a tabindex="0" href="#transactions" aria-selected="true" role="tab" class="v-tab v-tab--active" id="transactionsTab">Transactions</a> <a tabindex="0" href="#interactions" aria-selected="false" role="tab" class="v-tab" id="interactionsTab">Read / Write</a> <a tabindex="0" href="#holders" aria-selected="false" role="tab" class="v-tab" id="holdersTab">Holders</a> <a tabindex="0" href="#transfers" aria-selected="false" role="tab" class="v-tab" id="transfersTab">Transfers</a> <a tabindex="0" href="#analytics" aria-selected="false" role="tab" class="v-tab" style="display: none;" id="analyticsTab">Analytics</a>
        </div>
      </div>
      <div class="v-slide-group__next v-slide-group__next--disabled">
        <transition-stub name="fade-transition"></transition-stub>
      </div>
    </div>
  </div>
  <div class="v-window v-item-group theme--light v-tabs-items">
    <div class="v-window__container">
      <transition-stub name="v-window-x-transition">
        <div class="v-window-item v-window-item--active" id="transactions">
          <div class="container container--fluid">
            <div class="v-data-table v-data-table--has-bottom theme--light">
              <div class="v-data-table__wrapper">
                <table>
                  <colgroup>
                    <col class="">
                    <col class="">
                    <col class="">
                    <col class="">
                    <col class="">
                    <col class="">
                    <col class="">
                    <col class="">
                  </colgroup>
                  <thead class="v-data-table-header">
                    <tr>
                      <th role="columnheader" scope="col" aria-label="Txn Hash: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Txn Hash</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="Method" class="text-start"><span>Method</span></th>
                      <th role="columnheader" scope="col" aria-label="Block: Sorted descending. Activate to sort ascending." aria-sort="descending" class="text-start sortable active desc"><span>Block</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="Mined On: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Mined On</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="From: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>From</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="To: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>To</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="Value: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>Value</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
                      <th role="columnheader" scope="col" aria-label="Fee" class="text-start"><span>Fee</span></th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr class="v-data-table__empty-wrapper">
                      <td colspan="8">
                        No transactions found
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <div class="v-data-footer">
                <div class="v-data-footer__select">Rows per page:<div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted v-select">
                    <div class="v-input__control">
                      <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-101" class="v-input__slot">
                        <div class="v-select__slot">
                          <div class="v-select__selections">
                            <div class="v-select__selection v-select__selection--comma">10</div><input aria-label="Rows per page:" id="input-101" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                          </div>
                          <div class="v-input__append-inner">
                            <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                          </div><input type="hidden" value="10">
                        </div>
                        <div class="v-menu">
                          <!---->
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div class="v-data-footer__pagination">â€“</div>
                <div class="v-data-footer__icons-before"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Previous page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--light"></i></span></button></div>
                <div class="v-data-footer__icons-after"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Next page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light"></i></span></button></div>
              </div>
            </div>
          </div>
        </div>
      </transition-stub>
      <transition-stub name="v-window-x-transition">
        <!---->
      </transition-stub>
      <transition-stub name="v-window-x-transition">
        <!---->
      </transition-stub>
      <transition-stub name="v-window-x-transition">
        <!---->
      </transition-stub>
      <transition-stub name="v-window-x-transition">
        <!---->
      </transition-stub>
    </div>
  </div>
</div>
`;
